### DISCUSSION TITLE
Optionally export electrical field from bases\*.mph

### DISCUSSION THREAD

**Discussion by PALMS-gui** [wmglab-duke/ascent#22] [Ideas]
In order to visualize the field in 3rd party tools (Python, MATLAB), currently it is needed open each required base file and export solution: select the grid, output variables, filename and other options. 
![afbeelding](https://user-images.githubusercontent.com/72300527/160853612-d75b85b8-910c-4ea1-b441-e875b02fb99e.png)
![afbeelding](https://user-images.githubusercontent.com/72300527/160853665-1a40ffca-f66b-4234-a27d-aced514f107e.png)

It can also be (optionally) done via Java API.
Extracting Ve not only for fiber locations, but for arbitrary volume\surface would be a nice addition to the framework. 

Then it is possible to sum up active sources and get field distributions like that:
![afbeelding](https://user-images.githubusercontent.com/72300527/160855186-39c56711-bbd8-4ef8-865d-605f1d8b1ca1.png)

Then it is one step from making it dynamic (modulated by all kinds of waveforms) and replicating what is done by ASCENT in .hoc files.

**Comment by thedannymarsh** [comment]
Hi PALMS-gui,

Thanks for the feedback! We have discussed adding this feature. We may have the time to implement this in future releases, but would also welcome a pull request if you decide to implement it! It would involve adding a command line flag to run.json, something like "export_basis_Ve=true" and then adding the requisite Java code. A useful tool in this is COMSOL's application builder, where you can "record" a method, and COMSOL will generate the Java code to perform any actions you take in the GUI.

Thanks!
